src/client/api/init.c

```diff
@@ -55,6 +55,7 @@ const struct daos_task_api dc_funcs[] = {
 	{dc_pool_add_replicas, sizeof(daos_pool_replicas_t)},
 	{dc_pool_remove_replicas, sizeof(daos_pool_replicas_t)},
 	{dc_mgmt_list_pools, sizeof(daos_mgmt_list_pools_t)},
+	{dc_mgmt_get_bs_state, sizeof(daos_mgmt_get_bs_state_t)},
```

Comment:
@NiuYawei I was thinking this could just be solely used to query internal blobstore state for tests given the blobstore uuid. I don't think returning blobstore state at this time to the user would be necessary do you (via dmg)? I think the dc_mgmt_query_target would be a different effort/patch, that would be apart of dmg pool query command. With this pool query command with the --target specified we can return pool space info, and also possibly the VOS target state with that command?

Also, I want Samir to be able to use this blobstore state API asap for his verify device state test #3439

Reply:
dc_pool_query_target() is for per pool target query, I suppose dc_mgmt_query_target() could be used to query per server target.
Anyway, I'm fine with either 'per pool target query' or 'per server target query'. My point is that it's not necessary to introduce API, exported BIO interfaces to query bs state only, you could convert current dc_mgmt_get_bs_state() to a generic server target query API dc_mgmt_query_target() (so for the BIO exported interfaces), so that we can expand it later when necessary.